# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gcc/protector-adapter.diff
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

make __guard and __stack_smash_handler weak, so that they can be
overridden (dietlibc does this for example)

--- ./gcc_3_4_4.dif~	2005-08-22 01:03:26.000000000 -0400
+++ ./gcc_3_4_4.dif	2005-09-08 20:21:37.767333480 -0400
@@ -962,7 +962,7 @@
 + #endif
 + #endif
 + 
-+ long __guard[8] = {0, 0, 0, 0, 0, 0, 0, 0};
++ long __guard[8] __attribute ((weak)) = {0, 0, 0, 0, 0, 0, 0, 0};
 + static void __guard_setup (void) __attribute__ ((constructor));
 + 
 + static void
@@ -987,7 +987,7 @@
 + }
 + 
 + void
-+ __stack_smash_handler (char func[], ATTRIBUTE_UNUSED int damaged)
++ __attribute ((weak)) __stack_smash_handler (char func[], ATTRIBUTE_UNUSED int damaged)
 + {
 + #if defined (__GNU_LIBRARY__)
 +   extern char * __progname;
